src/mrpro/data/traj_calculators/_KTrajectoryCalculator.py

```diff
@@ -57,19 +58,21 @@ def _kfreq(self, kheader: KHeader) -> torch.Tensor:
         ValueError
             Center sample has to be the same for each readout
         """
-        n_samples = kheader.acq_info.number_of_samples
-        center_sample = kheader.acq_info.center_sample
-
-        # Verify that each readout has the same number of samples and same center sample
+        n_samples = kheader.acq_info.number_of_samples.flatten()
+        center_sample = kheader.acq_info.center_sample.flatten()
         if len(torch.unique(n_samples)) > 1:
             raise ValueError('Trajectory can only be calculated if each acquisition has the same number of samples')
-        if len(torch.unique(center_sample)) > 1:
-            raise ValueError('Trajectory can only be calculated if each acquisition has the same center sample')
+        n_k0 = int(n_samples[0])
+
+        # Data can be obtained with standard or reversed readout (e.g. bipolar readout).
+        k0 = torch.linspace(0, n_k0 - 1, n_k0, dtype=torch.float32)[None, :] - center_sample[:, None]
+        reversed_readout_idx = torch.where(kheader.acq_info.flags.flatten() & AcqFlags.ACQ_IS_REVERSE.value)[0]
+        # pytorch does not yet allow for negative indexing. torch.flip makes a copy of the data. providing a reversed
+        # index should be fastest
+        reversed_index = torch.linspace(n_k0 - 1, 0, n_k0, dtype=torch.int64)
+        k0[reversed_readout_idx, :] = k0[reversed_readout_idx[:, None], reversed_index[None, :]]
 
-        # Calculate points along readout
-        n_k0 = int(n_samples[0, 0, 0])
-        k0 = torch.linspace(0, n_k0 - 1, n_k0, dtype=torch.float32) - center_sample[0, 0, 0]
-        return k0
+        return k0.reshape(kheader.acq_info.center_sample.shape[:-1] + (n_k0,))
```

Comment:
https://docs.astral.sh/ruff/rules/collection-literal-concatenation/ does not yet catch this, as shape and the tuple are of differnt type, but you can even consider instead doing
` k0.reshape(*kheader.acq_info.center_sample.shape[:-1], n_k0)`
as reshape takes in integers

Reply:
see commit